app/enterprise/0.32-x/upgrades-migrations.md

```diff
@@ -0,0 +1,56 @@
+ For database backups you can use specific tools for each database such as `pg_dump` for Postgres or `nodetool snapshot` for Cassandra; however we recommend to always refer to an experienced DBA.
+
+Once you downloaded the package you review the change log to understand whether anything in your infrastructure needs to be upgraded consequentially, particular attention must be made around the supported database versions which may fall into a “no more supported state” due addition which Kong developers leverage to make queries and operation more efficient in the DAO and database level.
+Installing the newer package updates core libraries, plugins including a `kong.conf.default` which you can review to take into account newer or changes to configuration properties, those are usually documented in the changelog as well.
+Following the changelog itself is essential that you review any breaking changes and adapt your deployment process, kong configurations files, custom nginx template and anything else may be affected.
+
+**When performing upgrades in a solution with multiple kong nodes running, always shut down the other nodes and perform the upgrade operation one node at the time.**
+
+Running the `kong migrations up` command an update of the existing schema to the newer version is performed, old kong packages running on updated database schemas are not guaranteed to work correctly.
+At this stage, you want to validate the upgrade and smoke test your preproduction environment.
+One smart way of doing so is verifying the behaviour with the application of the canary plugin on a low priority api/service entity in preproduction redirecting it to correspondent entity in the upgraded production (if you setup allows doing so) and you may want to add also request termination plugin in the source environment, or doing so entirely with a reverse-proxy hard redirect for specifics low priority routes, if you have any in your infrastructure, nevertheless using the old school postman or curl commands.
+Once preproduction is validated you can go ahead and perform the same upgrade in production according to with your experience in preproduction following your business processes.
+
+## Notes for rollbacks
+Although Kong provides a `kong migrations down` command we suggest to rollback using a copy of the database taken before the upgrade (step 10), uninstalling the latest package version and reinstalling the previous version connected to the database with the old schema pre-upgrade.
+
+## Notes for Migrations to different infrastructure
+Indeed Kong is platform agnostic, however, there are other elements of the solutions that may differ between platform, specific examples can be Identity Providers, DNS servers etc.
```

Comment:
"differ between platform, specific examples..." -> "differ between platforms, including..."

Reply:
I don't accept this. You are asking me to change a word with a synonymous. This is not the scope of a technical review.

Reply:
This is a grammar improvement.